src/main/java/org/testng/internal/annotations/TestAnnotation.java

```diff
@@ -172,16 +172,16 @@ public IRetryAnalyzer getRetryAnalyzer() {
   }
 
   @Override
-  public void setRetryAnalyzer(Class<?> c) {
+  public void setRetryAnalyzer(Class<? extends IRetryAnalyzer> c) {

```

Comment:
@juherr - So here we have simplified the case, and gotten rid of the instance check. We still need to silently gobble down the inability to instantiate the class, because the default value is `Class` (Annotations doesn't accept null as a valid value) and `Class` cannot be instantiated. So if we threw exceptions here, then we would end up throwing exceptions for literally every case, wherein a user doesn't use a valid retry analyzer which kind of breaks backward compatibility in a big way.

Reply:
If you really want to be clean on this subject, the default retryAnalyzer class should be a real IRetryAnalyzer and we should warn the users if the one they provide is not an IRetryAnalyzer. It will allow us to prepare the change of the annotation definition (from Class to Class<? extends IRetryAnalyzer>).